README.md

```diff
@@ -479,6 +509,89 @@ registry.define('my-widget', () => {
 });
 ```
 
+#### Injecting State
+
+Working with larger widget structures, it can quickly become tiresome and complex to pass all the required properties down the tree. Additionally having to do so often means that widgets are forced to be aware of properties that they they don't need, but are required to know about in order to ensure they are propagated to the children.
+
+Dojo 2 provides a mechanism to inject state directly to parts of the widget structure; this is done by defining an `Injector` in the `registry` and passing a context object that will source the state that is to be injected.
+
+```ts
+import { Injector, BaseInjector } from '@dojo/widget-core/Injector';
+
+const myStateContext = {
+	theme: 'solid'
+};
+
+registry.define('state', Injector(BaseInjector, myStateContext));
+```
+
+To use the injected state, create a `beforeRender` method that returns a render function which creates a `w` reference to the registry item like any other widget and pass the properties required by the `InjectorProperties` interface.
```

Comment:
Maybe add an explanation of the above code, e.g. Here you can see we created a context called `myStateContext` and added a single state property `theme` to it, a gave this property the value of `solid`? or too tutorially?